### PR TITLE
Refactor f8_t and bf8_t as custom types, enable use of custom types

### PR DESCRIPTION
**Pull request by geyyer**
- Refactor f8_t and bf8_t
- Update conversion methods
- Update load method
- Add dynamic buffer custom types support
- Update threadwise conversion

Right now custom types are supported as source types, their support as destination types is coming soon...

### CODE REVIEW

File: include/ck/utility/data_type.hpp [hunk 1]
```diff
@@ -156,21 +178,1208 @@ struct scalar_type<bf8_t>
...
-    using d1_t = T;
-    using type = d1_t;
+    using type                           = _BitInt(8);
+    static constexpr index_t vector_size = 1;
+};
+
+int static err = 0;
+
+template <>
+struct vector_type<f8_t, 1>
```
Comment (zjing14) [inline include/ck/utility/data_type.hpp:191]: Could we template the vector_type instead of reimplementing it for each new data type?

**Comment by zjing14** [comment]
Could you do a performance check to make sure the new custom data type has no impact on the performance of fp8_gemm?

**Comment by geyyer** [comment]
@zjing14 There is no perf drop on example_gemm_xdl_fp8, nice!

**Comment by geyyer** [comment]
Current work is covered in https://github.com/ROCm/composable_kernel/pull/1333

**Pull request closed without merging**